app/controllers/users_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :check_admin, except: [:index]
+  def check_admin
+    if !current_user.has_role?(:admin)
+       redirect_to root_path
+    end
+  end
+  def role_adder(role)
```

Comment:
Try the method name `toggle_role` maybe? In general, method names should be a verb.